pkg/analyze/kube_resource.go

```diff
@@ -0,0 +1,172 @@
+package analyzer
+
+import (
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"strconv"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	iutils "github.com/replicatedhq/troubleshoot/pkg/interfaceutils"
+	"gopkg.in/yaml.v2"
+)
+
+func FindResource(kind string, namespace string, name string, getFileContents func(string) (map[string][]byte, error)) (interface{}, error) {
+
+	filemap := map[string]string{
+		"Deployment":           "deployments",
+		"StatefulSet":          "statefulsets",
+		"NetworkPolicy":        "network-policy",
+		"Pod":                  "pods",
+		"Ingress":              "ingress",
+		"Service":              "services",
+		"ResourceQuota":        "resource-quotas",
+		"Job":                  "jobs",
+		"PersistentVoumeClaim": "pvcs",
+		"pvc":                  "pvcs",
+		"ReplicaSet":           "replicasets",
+		"Namespace":            "namespaces.json",
+		"PersistentVolume":     "pvs.json",
+		"pv":                   "pvs.json",
+		"Node":                 "nodes.json",
+		"StorageClass":         "storage-classes.json",
+	}
+
+	var datapath string
+
+	resourceLocation, ok := filemap[kind]
+
+	if ok != true {
+		return nil, errors.New("failed to find resource")
+	}
+
+	if namespace != "" {
+		datapath = filepath.Join("cluster-resources", resourceLocation, fmt.Sprintf("%s.json", namespace))
+	} else {
+		datapath = filepath.Join("cluster-resources", resourceLocation)
+	}
+
+	files, err := getFileContents(datapath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read collected resources")
+	}
+
+	var resource interface{}
+	for _, file := range files {
+		err = yaml.Unmarshal(file, &resource)
```

Comment:
`resource` will always have the last successfully parsed objects. The rest will be ignored.

Reply:
The problem with removing the loop here is that `getFileContents` as used elsewhere in the code returns a `map[string][]byte` and I don't know what the `string` will be

Reply:
I think what you want here is to store all unmarshalled resources into a list and search for your selected item from all resources
```go
	resources := []interface{}{}
	for _, file := range files {
		... unmarshal resource and store
		resources = append(resources, resource)
	}

	for _, resource := range resources {
		items, err := iutils.GetAtPath(resource, "items")
		...

		itemslice := items.([]interface{})
		for _, item := range itemslice {
			...
			if name == name {
				return item, nil
			}
		}
	}

    # We did not find any item using the provided selector
    return nil, nil
```
Something like above